[BACKEND] Fix flaky test_financial_insights cached test (BCB market rates timeout)

## Bug

\`test_financial_insights_return_cached_period_without_provider_call\` falha em CI:

\`\`\`
FAILED tests/test_ai_advisory_service.py::TestAIAdvisoryServiceFinancialInsights::test_financial_insights_return_cached_period_without_provider_call - app.services.ai_advisory_service.AIInsightCostBudgetExceededError: Orçamento diário de AI Insights atingido (limite_usd=0.000001, gasto_usd=0.00003900).
\`\`\`

**Blocker:** test failing em master há dias, bloqueando todas as PRs api (1380, 1381, 1375).

## Root cause

O teste depende de cache HIT na segunda chamada de \`generate_financial_insights\`. Cache key inclui hash do snapshot, que inclui market rates obtidas do **BCB API real** (BcbMarketRatesProvider).

Quando o BCB API timeouta em CI:
- Primeira call: pode ter market rates fresh ou parcial → snapshot A → hash A
- Segunda call: timeout diferente → snapshot B → hash B
- Cache miss → vai para budget enforcement → \`AIInsightCostBudgetExceededError\` (porque o teste setou budget = \$0.000001)

Log do test mostra exatamente isso:
\`\`\`
WARNING app.services.market_rates_provider: ai_advisory.market_rates.fetch_failed series=4391 ym=2026-05 error=The read operation timed out
\`\`\`

**Não é um bug em \`ai_advisory_service.py\`** — o cache lookup ESTÁ antes do budget enforcement (linha 991-1004). É um problema de não-determinismo no test setup: usar provider real ao invés de stub.

## Fix

Adicionar em \`tests/conftest.py\`:

1. \`AI_MARKET_RATES_PROVIDER=stub\` em \`TEST_ENV_OVERRIDES\` → todos os testes usam o \`StubMarketRatesProvider\` determinístico
2. Autouse fixture \`reset_market_rates_provider_singleton\` que reseta o singleton antes/depois de cada teste — garante que o env var seja relido (singleton é process-level)

## Test plan

- [x] Test específico passa local com fix
- [x] 2388/2389 tests passam (1 skipped, 0 failed) na suite inteira local
- [ ] CI verde no PR

## Rollback

\`git revert\`. Volta a flaky comportamento anterior (que estava OK para alguns testes mas quebrava esse).

## Impacto

Desbloqueia 3 PRs em vôo:
- #1380 (RDS migration step 1)
- #1381 (RDS migration step 2)
- #1375 (governance retire)

E qualquer PR futuro do auraxis-api.

Refs: api#1376 (RDS migration umbrella)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -21,6 +21,12 @@ TEST_ENV_OVERRIDES = {
     "CORS_ALLOWED_ORIGINS": "https://frontend.local",
     "GRAPHQL_ALLOW_INTROSPECTION": "true",
     "BRAPI_CACHE_TTL_SECONDS": "0",
+    # Force the deterministic stub provider for market rates (BCB SGS data)
+    # so AI advisory tests don't flake on real BCB API timeouts. The default
+    # is BcbMarketRatesProvider which hits the public BCB API and returns
+    # different values on each call when the network is slow — that breaks
+    # snapshot hashing and cache-hit behaviour in tests.
+    "AI_MARKET_RATES_PROVIDER": "stub",
 }
 
 for key, value in TEST_ENV_OVERRIDES.items():
@@ -37,6 +43,29 @@ def isolate_test_env() -> Generator[None, None, None]:
             os.environ.pop(key, None)
         else:
             os.environ[key] = value
+
+
+@pytest.fixture(autouse=True)
+def reset_market_rates_provider_singleton() -> Generator[None, None, None]:
+    """Reset the BCB market-rates singleton so each test gets the stub.
+
+    The provider is a process-level singleton picked the first time it is
+    needed. If a test (or app import) instantiates it before our env var
+    `AI_MARKET_RATES_PROVIDER=stub` is read, subsequent tests inherit the
+    real BCB adapter and start flaking on timeouts. Resetting before each
+    test forces re-evaluation with our test env.
+    """
+    try:
+        from app.services.market_rates_provider import (
+            reset_market_rates_provider_for_tests,
+        )
+    except ImportError:
+        yield
+        return
+
+    reset_market_rates_provider_for_tests()
+    yield
+    reset_market_rates_provider_for_tests()
 
 
 @pytest.fixture

